sdk/lib/event_group/event_group.cc

```diff
@@ -8,6 +8,17 @@
 
 using Debug = ConditionalDebug<false, "Event groups library">;
 
+/**
+ * Length of a locking step for even group locks and futex words. When we
+ * attempt to grab an event group lock or wait on a futex, do so in steps of
+ * `EventGroupLockTimeoutStep`. That way, if the user of the event group
+ * crashes and we somehow do not have a pointer to the event group lock anymore
+ * (and thus cannot reset it), we have a guarantee that the blocking thread
+ * will become live again within a bounded amount of time, and crash as it is
+ * trying to re-acquire the lock if we free it through heap-free-all.
+ */
+static constexpr const Ticks EventGroupLockTimeoutStep = MS_TO_TICKS(500);
```

Comment:
It would be nice to make this configurable somehow.  Waking up every 500ms will really hurt power efficiency for a load of things that are mostly asleep.

Maybe make it a parameter and, in the FreeRTOS wrapper, expose a define for configuring it?  That way the network stack can set it to a smallish value.

Reply:
Sounds good, agree.